builder/build_unix_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// +build darwin dragonfly freebsd linux netbsd openbsd solaris
+
+package builder
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func Test_cp(t *testing.T) {
+	aModes := []int{
```

Comment:
I can't see why my comments are gone, but I suggested just using two modes here - we just need the minimum values to prove the point - that the number is honoured.